src/NLog/Layouts/JsonLayout.cs

```diff
@@ -113,6 +147,11 @@ public JsonLayout()
         public ISet<string> ExcludeProperties { get; set; }
 #endif
 
+        /// <summary>
+        /// How far should the JSON serializer follow object references before backing off
+        /// </summary>
+        public int MaxRecursionLimit { get; set; }
```

Comment:
is no max, -1 or should we make it nullable?

Reply:
I guess I can make it unassigned by having a nullable backing field. But until NLog 5.0, then it must have default value 0, or else it will be a breaking change.

Reply:
it;s a new property so not a breaking change?

Reply:
Yes it is a new property.

Reply:
Default 0 is fine

But I think we should change the type to nullable int, so we could disable the limit clearly. Agree?

Reply:
I think the limit should always be in place. Not sure if you have recognized, that the MaxRecursionLimit overrides the hardcoded value of 10 in NLog 4.5 BETA (And overrides the hardcoded value of 0 in NLog 4.4). The limit is there to protect the users from dumping the entire world into a single logevent.

Reply:
> What behavior do you want when having assigned MaxRecursionLimit to null ?

no limit.

Reply:
Think int.MaxValue will do just fine. A value higher than 10 will be a shotgun ready to shoot yourself. Actually considered an enum-value called NoReflection (=0), SingleReflection (=1), MaxReflection (=10), but thought that an integer would be easier. Now I see with this discussion, then it might have been a mistake.